core/src/main/java/com/datastrato/gravitino/tenant/SupportsUserManagement.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.tenant;
+
+import com.datastrato.gravitino.User;
+import com.datastrato.gravitino.exceptions.NoSuchUserException;
+import com.datastrato.gravitino.exceptions.UserAlreadyExistsException;
+import java.util.Map;
+
+/** The interface for the user management. */
+public interface SupportsUserManagement {
+
+  /**
+   * Creates a new User.
+   *
+   * @param metalake The Metalake of the User.
+   * @param userName THe name of the User.
+   * @param properties Additional properties for the Metalake.
+   * @return The created User instance.
+   * @throws UserAlreadyExistsException If a User with the same identifier already exists.
+   * @throws RuntimeException If creating the User encounters storage issues.
+   */
+  User createUser(String metalake, String userName, Map<String, String> properties);
```

Comment:
We use metalake and userName instead of identifier.  Because we don't express the layer relationship much. The User's identifier method is different from other class. It uses the field `metalake` and `userName` to create it.